client/rest/src/main/java/org/opensearch/client/RestClient.java

```diff
@@ -674,19 +674,7 @@ private HttpUriRequestBase addRequestBody(HttpUriRequestBase httpRequest, HttpEn
     static URI buildUri(String pathPrefix, String path, Map<String, String> params) {
         Objects.requireNonNull(path, "path must not be null");
         try {
-            String fullPath;
-            if (pathPrefix != null && pathPrefix.isEmpty() == false) {
-                if (pathPrefix.endsWith("/") && path.startsWith("/")) {
-                    fullPath = pathPrefix.substring(0, pathPrefix.length() - 1) + path;
-                } else if (pathPrefix.endsWith("/") || path.startsWith("/")) {
-                    fullPath = pathPrefix + path;
-                } else {
-                    fullPath = pathPrefix + "/" + path;
-                }
-            } else {
-                fullPath = path;
-            }
-
+            String fullPath = buildUriPath(pathPrefix, path);
```

Comment:
It seems like we just need to `return new URI("/").resolve(uriBuilder.build().toASCIIString());` to make sure URI is always resolved properly:

```
 new URI("/").resolve("/_render/template") ==> /_render/template
 new URI("/").resolve("_render/template") ==> /_render/template
```

Reply:
I ended up tweaking the `URIBuilder` construction to use its appending rather than building up the full path manually which then takes care of prefixing the `/`.